lib/digital_signature_api/crl/crl_service.ex

```diff
@@ -0,0 +1,173 @@
+defmodule DigitalSignature.CrlService do
+  @moduledoc false
+  use GenServer
+  use DigitalSignature.DateUtils
+  require Logger
+
+  alias DigitalSignature.CrlApi
+  alias DigitalSignature.Crl
+  alias DigitalSignature.RevokedSN
+
+  # Callbacks
+
+  @impl true
+  def init(state) do
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_info({:update, url}, state) do
+    newState = update_url_state(url, state)
+
+    {:noreply, newState}
+  end
+
+  @impl true
+  def handle_call({:update, url}, _from, state) do
+    newState = update_url_state(url, state)
+
+    {:reply, Map.has_key?(newState, url), newState}
+  end
+
+  def update_url_state(url, state) do
+    clean_timer(url, state)
+
+    with {:ok, nextUpdate} <- update_crl(url),
+         {:ok, nt} <- next_update_time(nextUpdate) do
+      tref = Process.send_after(__MODULE__, {:update, url}, nt)
+      Map.put(state, url, tref)
+    else
+      _ -> state
+    end
+  end
+
+  def start_link() do
+    started = GenServer.start_link(__MODULE__, %{}, name: __MODULE__)
+    timeout = Confex.fetch_env!(:digital_signature_api, DigitalSignature.CrlService)[:crl_process_timeout]
+
+    crl_urls()
+    |> Enum.each(fn url ->
+      try do
+        GenServer.call(__MODULE__, {:update, url}, timeout)
+      catch
+        error, reason ->
+          Logger.error(:io_lib.format("Error receiving ~s ~p : ~p", [url, error, reason]))
```

Comment:
I think You can just use `Logger.error("Error receiving #{url} #{inspect(error)} : #{inspect(reason)}")